lib/src/lints/dart/boolean_prefix.dart

```diff
@@ -74,6 +74,7 @@ class BooleanPrefix extends DartLintRule {
     context.registry.addMethodDeclaration((node) {
       final returnType = node.returnType?.type;
       if (returnType == null || !returnType.isDartCoreBool) return;
+      if (node.declaredElement?.hasOverride ?? true) return;
```

Comment:
I think this should be `if (node.declaredElement?.hasOverride ?? false)`.
I personally prefer `if (node.declaredElement?.hasOverride == true)`.
Using nullable boolean with `??` is sometimes confusing.

Reply:
> I think this should be `if (node.declaredElement?.hasOverride ?? false)`.

No this means if the declaredElement is null then the if block will be ignored and rest of the program will execute which we don't want if the declaredElement is null (ast is not yet resolved ).

> I personally prefer `if (node.declaredElement?.hasOverride == true)`.

This is also incorrect due to the same argument as above.

if you find it confusing we can write in this way

```dart
final elem=node.declaredElement;
if(elem == null || elem.hasOverride) return ;
```

Reply:
I see. Let's do it this way.
But use `element` instead of `elem`.